spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/secretmanager/GcpSecretManagerTemplateConfiguration.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2017-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.autoconfigure.secretmanager;
+
+import java.io.IOException;
+
+import com.google.api.gax.core.CredentialsProvider;
+import com.google.cloud.secretmanager.v1beta1.SecretManagerServiceClient;
+import com.google.cloud.secretmanager.v1beta1.SecretManagerServiceSettings;
+
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.cloud.gcp.core.DefaultCredentialsProvider;
+import org.springframework.cloud.gcp.core.DefaultGcpProjectIdProvider;
+import org.springframework.cloud.gcp.core.GcpProjectIdProvider;
+import org.springframework.cloud.gcp.core.UserAgentHeaderProvider;
+import org.springframework.cloud.gcp.secretmanager.SecretManagerTemplate;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+/**
+ * Autoconfiguration for GCP Secret Manager which provides an instance of the
+ * {@link SecretManagerTemplate}.
+ *
+ * @author Daniel Zou
+ * @since 1.2.2
+ */
+@Configuration
+@EnableConfigurationProperties(GcpSecretManagerProperties.class)
+@ConditionalOnClass(SecretManagerServiceClient.class)
+@ConditionalOnProperty(value = "spring.cloud.gcp.secretmanager.template.enabled", matchIfMissing = true)
+public class GcpSecretManagerTemplateConfiguration {
+
+	private final GcpSecretManagerProperties properties;
+
+	private final CredentialsProvider credentialsProvider;
+
+	private final GcpProjectIdProvider gcpProjectIdProvider;
+
+	public GcpSecretManagerTemplateConfiguration(
```

Comment:
`GcpSecretManagerTemplateConfiguration` and `GcpSecretManagerBootstrapConfiguration` end up being somewhat similar. 
I guess it's related to Mike's comment above, but possibly we should have one autoconfiguration with an optional conversion bean/property source additions that are blocked by an extra, off-by-default, property?

Is there any harm in doing regular bean initialization during bootstrap phase if someone does opt for template-only without enabling property initialization?

Reply:
Ohh I see, yes we can keep all beans in one configuration. I just created two because I wasn't sure if you could activate/deactivate beans by property as well.

So I could add a `@ConditionalOnProperty` to a bean definition?

Reply:
Discussed offline with @meltsufin; see: https://github.com/spring-cloud/spring-cloud-gcp/pull/2223#issuecomment-593570686